app/Http/Controllers/Api/V1/MetricsController.php

```diff
@@ -5,14 +5,22 @@
 use App\Http\Controllers\Controller;
 use Illuminate\Http\Response;
 use Illuminate\Http\Request;
+
 use App\Models\Activity;
 use App\Models\ActivityMetric;
 use App\Models\ActivityViewLog;
+use App\Models\Playlist;
+use App\Models\PlaylistMetric;
+use App\Models\PlaylistViewLog;
+use App\Models\Project;
+use App\Models\ProjectMetric;
+use App\Models\ProjectViewLog;
+
 
 class MetricsController extends Controller
 {
 
-    public function logview(Request $req, Activity $activity)
+    public function activity_logview(Request $req, Activity $activity)
     {
         $metrics = ActivityMetric::firstOrNew(
             ['activity_id' => $activity->id],
```

Comment:
Please use CamelCase for Function names (please correct it for other functions too)
Secondly, we are using a Repository Pattern instead of using model objects directly.
See ProjectRepository / ProjectRepositoryInterface as an example.
Please do this for other instances in the code where it is like that.

I am reporting this after discussion with Yuri. If this change is something that is blocker for other important functionality, then you can discuss with him, and we can leave it as is then.